espflash/src/connection/reset.rs

```diff
@@ -294,6 +346,29 @@ pub fn soft_reset(
     Ok(())
 }
 
+/// Perform Watchdog reset
+pub fn wdt_reset(chip: Chip, connection: &mut Connection) -> Result<(), Error> {
+    debug!("Resetting with RTC WDT");
+
+    let chip: Box<dyn RtcWdtReset> = match chip {
+        Chip::Esp32c3 => Box::new(crate::targets::esp32c3::Esp32c3),
+        Chip::Esp32s2 => Box::new(crate::targets::esp32s2::Esp32s2),
+        Chip::Esp32s3 => Box::new(crate::targets::esp32s3::Esp32s3),
+        _ => unreachable!(),
+    };
+
+    connection.write_reg(chip.wdt_wprotect(), chip.wdt_wkey(), None)?;
+    connection.write_reg(chip.wdt_config1(), 5000, None)?;
+    connection.write_reg(
+        chip.wdt_config0(),
+        (1 << 31) | (5 << 28) | (1 << 8) | 2,
+        None,
+    )?;
+    connection.write_reg(chip.wdt_wprotect(), 0, None)?;
```

Comment:
Can we add some links/pointer to make this less black magic?

Reply:
Probably wouldn't be a terrible idea to use [bitflags](https://docs.rs/bitflags/latest/bitflags/) for this, probably some other places we should be using this as well.